src/NServiceBus.Core/Installation/InstallerComponent.cs

```diff
@@ -0,0 +1,68 @@
+#nullable enable
+namespace NServiceBus;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Installation;
+using Microsoft.Extensions.DependencyInjection;
+using Settings;
+
+class InstallerComponent(InstallerComponent.Settings settings)
+{
+    public void Initialize(IReadOnlySettings globalSettings) => globalSettings.AddStartupDiagnosticsSection("Installation", new { InstallersEnabled = settings.Installers.Select(i => i.Name).ToArray() });
+
+    public async Task RunInstallers(IServiceProvider serviceProvider, CancellationToken cancellationToken = default)
+    {
+        foreach (var installer in settings.Installers)
+        {
+            await installer.Install(serviceProvider, settings.InstallationUserName, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    public class Settings
+    {
+        public string InstallationUserName { get; set; } = Environment.OSVersion.Platform == PlatformID.Win32NT ? $"{Environment.UserDomainName}\\{Environment.UserName}" : Environment.UserName;
+
+        public void Add<TInstaller>() where TInstaller : class, INeedToInstallSomething => installers.Add(new Installer<TInstaller>());
+
+        public void AddScannedInstallers(IEnumerable<Type> scannedTypes)
+        {
+            foreach (var installerType in scannedTypes.Where(IsINeedToInstallSomething))
+            {
+                var installerWrapperType = typeof(Installer<>).MakeGenericType(installerType);
+                installers.Add(((IInstaller)Activator.CreateInstance(installerWrapperType)!)!);
+            }
+        }
+
+        public IReadOnlyCollection<IInstaller> Installers => installers;
+
+        readonly HashSet<IInstaller> installers = [];
+
+        static bool IsINeedToInstallSomething(Type t) => typeof(INeedToInstallSomething).IsAssignableFrom(t);
+
+        class Installer<T> : IInstaller where T : class, INeedToInstallSomething
+        {
+            public Task Install(IServiceProvider serviceProvider, string identity, CancellationToken cancellationToken = default)
+            {
+                var installer = ActivatorUtilities.CreateInstance<T>(serviceProvider);
+
+                return installer.Install(identity, cancellationToken);
+            }
+
+            public bool Equals(IInstaller? other) => other?.GetType() == typeof(T);
```

Comment:
Why not compare the name?

Reply:
Wouldn't that be an issue with two installers with the same name but in different assemblies exists? (or are you talking about the assembly qualified name)

Reply:
I saw we expose a name which points to the full name. This means when we add this to diagnostics your example would lead to one overriding the other at least in the diagnostics. That's why I was wondering why we are not using the Name instead here. I also don't think doing other.GetType works because you get the type of the generic installer but you want to get the type of the T type. So in this case the installer interface would need to expose the underlying installer type for this comparison to be correct, no?